### PR TITLE
User schedule notifications improvement

### PR DESCRIPTION
**Pull request by dzonidoo**
CPCN-297

### CODE REVIEW

**Comment by petrjasek** [comment]
imo it would be better to use rather selects, 1 for hours and second for minutes (with just `00, 15, 30, 45` and `00` being the default value), atm it would allow you to select anything and then on save it complains but might be browser specific actually

**Comment by tomaskikutis** [comment]
I did think about that, but decided to suggest the time picker. One reason is out of the box 12/24 hour support and another is that it's less changes since the settings aren't going to be changed that often anyway. We could add a info message to specify the limitation on minutes. We could also reimplement to not use time picker if you think it's worth it.

**Comment by petrjasek** [comment]
ok we can run it by qa and we will see

File: assets/user-profile/components/EditNotificationScheduleModal.tsx [hunk 1]
```diff
@@ -49,13 +53,36 @@ class EditNotificationScheduleModalComponent extends React.Component<IProps, ISt
         });
     }
 
+    onSubmitForm(event: React.FormEvent<HTMLFormElement>) {
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (this.formRef.current == null) {
+            return;
```
Comment (tomaskikutis) [inline assets/user-profile/components/EditNotificationScheduleModal.tsx:61]: it's better to use `throw new Error('ref missing')` so it's easier to distinguish code that is not supposed to run at all

File: assets/user-profile/components/EditNotificationScheduleModal.tsx [hunk 2]
```diff
@@ -69,28 +96,37 @@ class EditNotificationScheduleModalComponent extends React.Component<IProps, ISt
...
+                                step="900"
+                                min="00:00"
                             />
                             <input
                                 type="time"
                                 value={this.state.times[1]}
                                 onChange={(event) => {
                                     this.updateTime(event.target.value, 1);
                                 }}
+                                step="900"
```
Comment (tomaskikutis) [inline assets/user-profile/components/EditNotificationScheduleModal.tsx:119]: move those props into a variable and put at the top of the file

**Review by petrjasek — APPROVED** [review]

**Pull request merged**